### PR TITLE
change to well posed NeoHooke

### PR DESCRIPTION
**Pull request by koehlerson**
https://github.com/Ferrite-FEM/Ferrite.jl/issues/844

### CODE REVIEW

**Comment by koehlerson** [comment]
image
looks quite the same, but `norm(u)` is slightly different, probably to be expected?

**Comment by koehlerson** [comment]
alternatively in https://onlinelibrary.wiley.com/doi/full/10.1002/nme.4351 another formulation is used which is not normalized. The current one from Wikipedia is normalized (in the sense of $\psi(\boldsymbol{1})=0$)

**Comment by KnutAM** [comment]
In true Ferrite-spirit: https://www.dealii.org/current/doxygen/deal.II/step_44.html#NeoHookeanmaterials and ref to Holzapfel (2001) therein :)
Perhaps nice to comment in the example that there are different possible formulations for the volumetric part (as is done in the deal.ii example)?

**Comment by koehlerson** [comment]
Citing a 400 pages book by "see Holzapfel (2001)" is an engineering practice, I'll never understand. The reference is equation (6.138). Within the book, the actual reference is Simo and Miehe (1992) (https://www.sciencedirect.com/science/article/pii/004578259290170O equation 2.37) . The chapter of equation (6.138) talks about "ogden model for compressible rubber-like materials". I can add later a reference to Simo and Miehe for one alternative choice and maybe another statement that this is free of choice as long as it is strictly convex in $J$

**Comment by KnutAM** [comment]
First of all: Sorry if I made you do a lot of digging now - that was not my intention! 

I think you should implement the (strictly convex, or polyconvex if that suffices) free energy that you prefer. (I agree that Psi(I)=0 is nice). I just thought that the following could be nice

1) Referring to where the model we use is taken from (doesn't have to be the original reference if well-established, could refer to an equation in a book)
2) Commenting that multiple options for the volumetric part exist/can be made (but additional examples/refs are not required IMO as it isn't a theory manual.)

**Comment by koehlerson** [comment]
> First of all: Sorry if I made you do a lot of digging now - that was not my intention!

Sorry if the message was perceived that way. It was meant more in the direction that I'd rather see in our docs some precise citation, especially since this tutorial may be the template for beginners (including me back then). The rant was more about the general situation in engineering publications where I encounter often the same.

> 
> I think you should implement the (strictly convex, or polyconvex if that suffices) 

if it's strictly convex in $J$ then $\psi(\boldsymbol{F})$ becomes polyconvex w.r.t. $\boldsymbol{F}$ which is a key ingredient of well posedness. That's why you want the strictly convex term in $J$.

> 
>     1. Referring to where the model we use is taken from (doesn't have to be the original reference if well-established, could refer to an equation in a book)

Yeah that's fine for me. Will do that tomorrow.

> 
>     2. Commenting that multiple options for the volumetric part exist/can be made (but additional examples/refs are not required IMO as it isn't a theory manual.)

Yeah I was planning some `note !!!` or the like

**Comment by lijas** [comment]
Perhaps this is a stupid questions, but does not chaning the potentional mean that it is no longar a Neo-hookean model? Or are there many types neo-hookean models? 
Also, the original formulation is only breaks down for really large deformation (J=10), right?

**Comment by KnutAM** [comment]
> Perhaps this is a stupid questions, but does not chaning the potentional mean that it is no longar a Neo-hookean model? Or are there many types neo-hookean models?

To my understanding, the true/original "NeoHooke" model is incompressible, but can be extended to a compressible formulation for which there are different variants of the volumetric contribution.

**Comment by lijas** [comment]
Ok thanks for clarifying. The original NeoHooke model would then require to be used with taylor-hood (displacement preasure) elements, I guess?

**Comment by koehlerson** [comment]
> Ok thanks for clarifying. The original NeoHooke model would then require to be used with taylor-hood (displacement preasure) elements, I guess?

Neo Hooke is just $C_1 (I_1 - 3)$ and if you compute this without a pressure formulation you can increase the volume freely leading to many (infinite I guess) solutions. So either a pressure field is incorporated for the incompressible model or you add some well posed penalty terms as we have in the example here. But it's definitely worth to mention that there is no such thing as "the" compressible Neo Hooke model

**Review by termi-official — CHANGES_REQUESTED** [review]
Hyperelasticity docs are broken on the deploy. I left a commen where this might come from. Can you check locally?

File: docs/src/literate-tutorials/hyperelasticity.jl [hunk 1]
```diff
@@ -64,11 +64,23 @@ using Ferrite, Tensors, TimerOutputs, ProgressMeter, IterativeSolvers
...
-# \Psi(\mathbf{C}) = \frac{\mu}{2} (I_1 - 3) - \mu \ln(J) + \frac{\lambda}{2} \ln(J)^2,
+# \Psi(\mathbf{C}) = \underbrace{\frac{\mu}{2} (I_1 - 3)}_{W(C)} \underbrace{- {\mu} \ln(J) + \frac{\lambda}{2} (J - 1)^2}_{U(J)},
 # ```
 #
 # where ``I_1 = \mathrm{tr}(\mathbf{C})`` is the first invariant, ``J = \sqrt{\det(\mathbf{C})}``
 # and ``\mu`` and ``\lambda`` material parameters.
+# Note that we use here a compressible formulation of the original incompressible Neo Hooke model $W(F)$.
+# In order to compensate the compressible behavior, a volume penalty $U(J)$ is incorporated in $\psi$.
+# !!! note
+#    The Neo-Hooke model is only a well defined terminology in the incompressible case.
```
Comment (termi-official) [inline docs/src/literate-tutorials/hyperelasticity.jl:75]: I think you need one more space here for the indentation.

**Review by KnutAM — COMMENTED** [review]
Great job fixing this and adding more details on neo-Hookean models with proper references!

**Pull request merged**